### PR TITLE
Begining C++23: Chapter 8

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Examples/Modules/Chapter 08/Ex8_18.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Constant expressions 
+
+import std;
+
+// Uncomment constexpr to make the definitions of a8 and p compile
+/*constexpr*/ std::size_t power(std::size_t x, unsigned n)
+{
+	return n == 0 ? 1 : x * power(x, n - 1);
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 08/Ex8_18.cpp:8]: You usually use 2 spaces instead of tabs for indentation.

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/Modules/Chapter 08/Ex8_07.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Modifying the value of a caller variable � references vs pointers
-import <iostream>;
+import std;
 
 void change_it_by_pointer(double* reference_to_it);    // Pass pointer (by value)
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 08/Ex8_07.cpp:4]: Nothing technically about it, that was a full blown mistake...

File: Exercises/Modules/Chapter 08/Soln8_06.cpp [hunk 3]
```diff
@@ -49,15 +44,15 @@ int main()
...
     int number{};
     std::cin >> number;
     if (number < 0)
       break;
     else if (number > 100)
-      std::cout << "Only numbers < 100, please..." << std::endl;
+      std::println("Only numbers < 100, please...");
     else
       grades.push_back(number);
   }
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Exercises/Modules/Chapter 08/Soln8_06.cpp:58]: I have passed fixed-size arrays on several occasions. But it's rare. I've turned it into a bonus exercise: that way we have both, an exercise on passing arrays by reference, and one to modernize old, crappy code. Plus, it's less work for me now ;-)

File: Examples/Modules/Chapter 08/Ex8_18.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+int main()
+{
+	std::array<int, 1> a1;                // Ok: literals are constant expressions
+
+	const std::size_t integer_constant{ 2 };
+	std::array<int, integer_constant> a2; // Ok: integer constants are constant expressions
+
+	enum class Size { ONE = 1, TWO = integer_constant, THREE }; // Ok: integer constant
+	std::array<int, static_cast<std::size_t>(Size::THREE)> a3;  // Ok: enumerator
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 08/Ex8_18.cpp]: Yeah, but then it becomes a function call, and I'm only explaining function calls later in the current narrative... So I'll pass... Could add a note later on, but it 'd only be distracting...

**Pull request merged**